cluster/gce/gci/configure.sh

```diff
@@ -254,7 +254,7 @@ function install-gci-mounter-tools {
   mkdir -p "${CONTAINERIZED_MOUNTER_HOME}"
   chmod a+x "${CONTAINERIZED_MOUNTER_HOME}"
   mkdir -p "${CONTAINERIZED_MOUNTER_HOME}/rootfs"
-  download-or-bust "${mounter_tar_sha}" "https://storage.googleapis.com/kubernetes-release/gci-mounter/mounter.tar"
+  download-or-bust "${mounter_tar_sha}" "https://dl.k8s.io/gci-mounter/mounter.tar"
```

Comment:
cc @msau42 we also really need to replace this never-updated tarball at some point

Reply:
We use this for testing nfs in gce cos jobs. If there's an alternate job on another OS, we could just disable nfs testing on gce cos. Looking at our sig-storage testgrid, we're only tracking gce and kind jobs, and IIRC kind had trouble with supporting nfs? https://testgrid.k8s.io/sig-storage-kubernetes

Reply:
kops jobs also run NFS tests